_rules/first-focusable-elements-are-links-to-content-e53727.md

```diff
@@ -1,159 +1,356 @@
+			<li><a href="#main">Skip to main content</a></li>
+		</ul>
+		<!-- Repeated Content -->
+		<nav id="navigation">Navigation section</nav>
+		<form role="search" id="search">Search section</form>
+		<main id="main">Main content</main>
+	</body>
+</html>
+```
+
+#### Passed Example 2
+
+The list of links are visible when focused.
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title></title>
+		<style>
+			#skiplinks a {
+				position: absolute;
+				top: 100px;
+			}
+			#skiplinks a:focus {
+				top: 0px;
+			}
+		</style>
+	</head>
+	<body>
+		<ul>
+			<li><a href="#navigation">Skip to navigation</a></li>
+			<li><a href="#search">Skip to search</a></li>
+			<li><a href="#main">Skip to main content</a></li>
+		</ul>
+		<!-- Repeated content -->
+		<nav id="navigation">Navigation section</nav>
+		<form role="search" id="search">Search section</form>
+		<main id="main">Main content</main>
+	</body>
+</html>
+```
+
+#### Passed Example 3
+
+The list of links when activated move focus to heading elements.
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title></title>
+	</head>
+	<body>
+		<ul>
+			<li><a href="#firstheading">First heading</a></li>
+			<li><a href="#secondheading">Second heading</a></li>
+		</ul>
+		<!-- Repeated Content -->
+		<h2 id="firstheading">First heading section</h2>
+		<!-- Content -->
+		<h2 id="secondheading">Second heading section</h2>
+		<!-- Content -->
+	</body>
+</html>
+```
+
+#### Passed Example 4
+
+The list of links use `aria-label` to provide an accessible name.
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title></title>
+	</head>
+	<body>
+		<ul>
+			<li><a href="#navigation" aria-label="Skip to navigation"></a></li>
+			<li><a href="#search" aria-label="Skip to search"></a></li>
+			<li><a href="#main" aria-label="Skip to main content"></a></li>
+		</ul>
+		<!-- Repeated content -->
+		<nav id="navigation">Navigation section</nav>
+		<form role="search" id="search">Search section</form>
+		<main id="main">Main content</main>
+	</body>
+</html>
+```
+
+#### Passed Example 5
+
+The list of links use `aria-describedby` to provide an accessible name.
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title></title>
+	</head>
+	<body>
+		<ul>
+			<li><a href="#navigation" aria-describedby="navigationHeading">Skip to navigation</a></li>
+			<li><a href="#search" aria-describedby="searchHeading">Skip to search</a></li>
+			<li><a href="#main" aria-describedb="mainHeading">Skip to main content</a></li>
+		</ul>
+		<!-- Repeated content -->
+		<nav id="navigation">
+			<h2 id="navigationHeading">Navigation section</h2>
+		</nav>
+		<form role="search" id="search">
+			<h2 id="searchHeading">Search section</h2>
+		</form>
+		<main id="main">
+			<h2 id="mainHeading">Main content</h2>
+		</main>
+	</body>
+</html>
+```
+
+### Failed
+
+#### Failed Example 1
+
+First focusable elements in the document do not have a semantic role of link.
```

Comment:
Here also, it could be good to have them move focus correctly but without being links.
It can also be with one of them not being a link and the rest being (only one bad is enough to fail).

Reply:
Great. I added another failing example for the use case of one of the links not having a semantic role of link. I could see needing more examples where one of the links doesn't have an accessible name, does not move focus to a section etc but that could get messy. Thoughts?

Reply:
Yes, w do need failing tests that fail for each individual reason. Here, Expectation 1 has 5 subconditions, so we need (at least) 5 tests, failing each one of these reason and none of the others.

When checking implementations of this rule, we need to be able to check that the implementation is actually doing the correct thing. If all failing examples lack, say, the first subcondition, then we could dub as "good" an implementation which only checks this subcondition, and that would be incorrect…

This is comparable with code like `if (a(x) && b(x) && c(x) && d(x) && e(x))`. Assuming that `a`, `b`, `c`, `d`, and `e` are not called elsewhere, the easiest way to ensure full coverage (including correct branching) is to have 5 tests, each failing one of these. If, say, all the test fail `e(x)`, then we cannot make the difference between `… && e(x)` and `… || e(x)`.

So, yes, we do need failing test case for each individual (sub)condition.